### PR TITLE
v4 hamburger icon fix

### PR DESCRIPTION
**Pull request by patrickhlauke**
closes https://github.com/twbs/bootstrap/issues/20234

### CODE REVIEW

**Comment by patrickhlauke** [comment]
comparison before/after screenshot (admittedly the `\2261` glyph is a bit anemic - hence the bump in `font-weight` - and not quite as wide as I'd like...)

> hamburglar

**Comment by patrickhlauke** [comment]
@mdo @cvrebert thoughts?

File: scss/_navbar.scss [hunk 1]
```diff
@@ -115,6 +115,12 @@
   @include hover-focus {
     text-decoration: none;
   }
+
+  &::before {
+    font-weight: 500;
+    content: "\2261"; // see http://unicode.johnholtripley.co.uk/2261/
```
Comment (cvrebert) [inline scss/_navbar.scss]: Maybe comment it as `IDENTICAL TO (U+2261)` similar to https://github.com/twbs/bootstrap/blob/1347cb0152c992f8b71e185fa7ef1794fc94d931/scss/_carousel.scss#L145?

File: scss/_navbar.scss [hunk 2]
```diff
@@ -115,6 +115,12 @@
   @include hover-focus {
     text-decoration: none;
   }
+
+  &::before {
```
Comment (cvrebert) [inline scss/_navbar.scss]: A comment explaining what this is for (e.g. `// Hamburger menu icon`) would be good.

**Comment by cvrebert** [comment]
I'd also suggest adding a comment explaining why we can't use TRIGRAM FOR HEAVEN (U+2630).
Documenting these sorts of design decisions / historical context is best-practice.

File: scss/_navbar.scss [hunk 3]
```diff
@@ -116,9 +116,15 @@
...
   }
 
+  // Hamburger menu icon
   &::before {
     font-weight: 500;
-    content: "\2261"; // see http://unicode.johnholtripley.co.uk/2261/
+    content: "\2261"; // IDENTICAL TO (U+2261)
+    // wider overall support (see http://unicode.johnholtripley.co.uk/2261/)
+    // compared to previously used TRIGRAM FOR HEAVEN (U+2630)
+    // using generated content, rather than adding this in markup,
```
Comment (cvrebert) [inline scss/_navbar.scss]: Erm, where did we establish that using generated content made any difference?

Comment (patrickhlauke) [inline scss/_navbar.scss]: Good call, i got confused

**Comment by cvrebert** [comment]
LGTM technically speaking. I can't speak as to the visual design; @mdo ought to weigh in.

**Comment by mdo** [comment]
Aesthetically that looks a bit off right now. I'd be curious to see an exploration that use an SVG background image like we do with our validation icons for comparison.

**Comment by patrickhlauke** [comment]
rough sketch of how an SVG approach may look like https://jsfiddle.net/patrick_h_lauke/nu5ejdy3/2/

**Comment by cvrebert** [comment]
A survey of implementation options: https://css-tricks.com/three-line-menu-navicon/

**Comment by patrickhlauke** [comment]
tricks using borders, shadows etc just feel...extremely hacky and old-school to me. my personal preference would be either an SVG or unicode approach (I guess SVG would give us more flexibility in design of course, so not adverse to changing this to match my rough fiddle https://jsfiddle.net/patrick_h_lauke/nu5ejdy3/3/ ... @mdo thoughts?)

**Comment by XhmikosR** [comment]
I agree with @mdo that visually the new one doesn't look so good.

The SVG one does look a lot better IMO.

**Comment by patrickhlauke** [comment]
@mdo @cvrebert @XhmikosR should i update this PR to take the approach i tried in my jsfiddle instead then? I'm tending towards that one too myself

**Comment by XhmikosR** [comment]
Personally I think yes. But wait for the others to confirm too.

**Comment by cvrebert** [comment]
Sounds like we're onboard with SVG. I'd say go for it Patrick.

**Comment by patrickhlauke** [comment]
closing in favor of successor https://github.com/twbs/bootstrap/pull/20329 which uses SVG

**Pull request closed without merging**